### DISCUSSION TITLE
Improve component documentation with use information and visuals

### DISCUSSION THREAD

**Discussion by TYRANTHOSAURUS** [resend/react-email#1333] [Ideas]
### Goals

1. Instruct users on the capabilities of each individual component so they know how to build their own template better
2. Provide quick building blocks for users so they can select and copy different styled components to their template from docs.
3. Provide information about components props, API so users know how to use them.

### Background

It's very tough to understand some components as a beginner such as Head, Code-Inline etc. There's no documentation at all what this does, why we need it or how we should apply it.

![Uploading Screenshot 2024-02-28 at 08.32.10.png…]()
![Uploading Screenshot 2024-02-28 at 08.32.43.png…]()

### Proposal

1. Show examples ready to use buttons users can copy in their template
2. Display differences in props, styling
3. Add API documentation

Example: Antd
![Uploading Screenshot 2024-02-28 at 08.33.38.png…]()

**Comment by gabrielmfern** [comment]
Seems like your screenshots didn't load in properly onto github
![image](https://github.com/resend/react-email/assets/88866334/970fb670-2e41-4810-9e88-088f1ce59c39)

**Comment by gabrielmfern** [comment]
Having an API reference documenting what we do to the props, or what changes we make would be amazing, and certainly a crazy improvement on the DX. 

I would to love hear some more suggestions from you if you have any on this, I'm working on some other improvements to the docs as well and I'll try to squeeze this in for the more complicated components in a few days.
